resource-managers/kubernetes/core/src/main/scala/org/apache/spark/shuffle/KubernetesLocalDiskShuffleExecutorComponents.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle
+
+import java.io.File
+import java.util.Optional
+
+import scala.reflect.ClassTag
+
+import org.apache.spark.{SparkConf, SparkEnv}
+import org.apache.spark.internal.Logging
+import org.apache.spark.shuffle.api.{ShuffleExecutorComponents, ShuffleMapOutputWriter, SingleSpillShuffleMapOutputWriter}
+import org.apache.spark.shuffle.sort.io.LocalDiskShuffleExecutorComponents
+import org.apache.spark.storage.{BlockId, BlockManager, StorageLevel, UnrecognizedBlockId}
+import org.apache.spark.util.Utils
+
+class KubernetesLocalDiskShuffleExecutorComponents(sparkConf: SparkConf)
+  extends ShuffleExecutorComponents with Logging {
+
+  private val delegate = new LocalDiskShuffleExecutorComponents(sparkConf)
+  private var blockManager: BlockManager = _
+
+  override def initializeExecutor(
+      appId: String, execId: String, extraConfigs: java.util.Map[String, String]): Unit = {
+    delegate.initializeExecutor(appId, execId, extraConfigs)
+    blockManager = SparkEnv.get.blockManager
+    if (sparkConf.getBoolean("spark.kubernetes.driver.reusePersistentVolumeClaim", false)) {
+      // Turn off the deletion of the shuffle data in order to reuse
+      blockManager.diskBlockManager.deleteFilesOnStop = false
+      logError("Recover shuffle data")
```

Comment:
This doesn't always recover shuffle data, right? Once `KubernetesLocalDiskShuffleDataIO` is used, `initializeExecutor` will be called per executor, even it is initial executor without failure.

Reply:
Yep. Right. This is a left-over of my debugging message. I used `logError`. :)